### PR TITLE
Added ability to build assets using Vite as an alternative to Webpack encore.

### PR DESCRIPTION
**Pull request by mickverm**
#### 🔧 Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Codebase improvement
- [ ] Other (if none of the other choices apply)

#### 🚨 Checklist

- [x] I've read the [guidelines for contributing](https://github.com/gremo/symfony-sail/blob/main/CONTRIBUTING.md)
- [x] I've added necessary documentation (if appropriate)
- [x] I've ensured that my code additions do not fail linting or unit tests (if applicable)
- [x] I've ensured that all commits' message styles match [semantic commits](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716)

#### Description

In my personal project, I'm using https://github.com/lhapaipai/vite-bundle and https://github.com/lhapaipai/vite-plugin-symfony instead of Webpack encore. This change builds assets via Vite when Webpack encore isn't available.

### CODE REVIEW

**Comment by gremo** [comment]
Thank you @mickverm! I don't know Vite but it seems a nice addition. I'll review ASAP.

File: Dockerfile [hunk 1]
```diff
@@ -44,13 +44,16 @@ RUN \
 
 # Copy the entire source tree. We need this because some tools, like Tailwind
 # CSS, scan source files (like templates) and extract class names in order to
-# purge unsed CSS.
+# purge unused CSS.
```
Comment (gremo) [inline Dockerfile]: Plese move the typo fix to another PR thanks

**Comment by gremo** [comment]
Seems fine to me. Please update README.md Key features list and the general introduction like:

> ... It includes support for Visual Studio Code Dev Containers, MariaDB and Webpack Encore/Vite.

**Comment by mickverm** [comment]
I've made the requested changes.

**Comment by gremo** [comment]
Thank you @mickverm
